Clarifies exit status information. Thanks to "A. Costa" <agcosta@gis.net>
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=446854

--- a/doc/grep.in.1
+++ b/doc/grep.in.1
@@ -1153,23 +1153,15 @@
 is not set.
 .
 .SH "EXIT STATUS"
-Normally, the exit status is 0 if selected lines are found and 1 otherwise.
-But the exit status is 2 if an error occurred, unless the
+The exit status is 0 if selected lines are found, and 1 if not found.
+If an error occurred (and
 .B \-q
 or
 .B \-\^\-quiet
 or
 .B \-\^\-silent
-option is used and a selected line is found.
-Note, however, that \s-1POSIX\s0 only mandates, for programs such as
-.BR grep ,
-.BR cmp ,
-and
-.BR diff ,
-that the exit status in case of error be greater than 1;
-it is therefore advisable, for the sake of portability,
-to use logic that tests for this general condition
-instead of strict equality with\ 2.
+are not used) the exit status is 2. (Note: POSIX error
+handling code should check for '2' or greater.)
 .
 .SH COPYRIGHT
 Copyright 1998-2000, 2002, 2005-2014 Free Software Foundation, Inc.
--- a/doc/grep.texi
+++ b/doc/grep.texi
@@ -1023,16 +1023,10 @@
 @cindex exit status
 @cindex return status
 
-Normally, the exit status is 0 if selected lines are found and 1 otherwise.
-But the exit status is 2 if an error occurred, unless the @option{-q} or
-@option{--quiet} or @option{--silent} option is used and a selected line
-is found.
-Note, however, that POSIX only mandates,
-for programs such as @command{grep}, @command{cmp}, and @command{diff},
-that the exit status in case of error be greater than 1;
-it is therefore advisable, for the sake of portability,
-to use logic that tests for this general condition
-instead of strict equality with@ 2.
+The exit status is 0 if selected lines are found, and 1 if not found.  If
+an error occurred (and @option{-q} or @option{--quiet} or
+@option{--silent} are not used) the exit status is 2. (Note: POSIX error
+handling code should check for '2' or greater.
 
 
 @node grep Programs
